src/kaltura-player.js

```diff
@@ -1191,6 +1191,124 @@ class KalturaPlayer extends FakeEventTarget {
   get remotePlayerManager() {
     return this._remotePlayerManager;
   }
+
+  /**
+   * get the media capabilities
+   * @function getMediaCapabilities
+   * @param {HEVCConfigObject} hevcConfig - The HEVC configuration to check (optional).
+   * @returns {Promise<MediaCapabilitiesObject>} - The media capabilities object.
+   * @public
+   */
+  async getMediaCapabilities(hevcConfig?: HEVCConfigObject): Promise<MediaCapabilitiesObject> {
+    let mediaCapabilities;
+    try {
+      KalturaPlayer._logger.debug('Starting to get media capabilities...');
+      const hevcSupported = await this._checkHEVCSupported(hevcConfig);
+      const drmSupported = await this._checkDRMSupported();
+      mediaCapabilities = Object.assign({}, hevcSupported, drmSupported);
+      KalturaPlayer._logger.debug('Finished getting media capabilities ', {mediaCapabilities});
+      return mediaCapabilities;
+    } catch (ex) {
+      KalturaPlayer._logger.debug('There was a problem with getting the media capabilities, ', ex.message);
+      mediaCapabilities = {
+        isHEVCSupported: false,
+        isPowerEfficient: false,
+        isDRMSupported: false,
+        supportedDRMs: []
+      };
+      KalturaPlayer._logger.debug('Returning media capabilities defaults ', {mediaCapabilities});
+      return mediaCapabilities;
+    }
+  }
+
+  /**
+   * checks whether and which DRMs are supported by the browser
+   * @function _checkDRMSupported
+   * @returns {Promise<DRMSupportedObject>} - The DRM supported object.
+   * @private
+   */
+  async _checkDRMSupported(): Promise<DRMSupportedObject> {
+    let drmSupportedRes: DRMSupportedObject = {
+      isDRMSupported: 'maybe',
+      supportedDRMs: []
+    };
+
+    if (!navigator.requestMediaKeySystemAccess) {
+      return drmSupportedRes;
+    }
+
+    const keySysConfig = [
+      {
+        initDataTypes: ['cenc'],
+        videoCapabilities: [
+          {
+            contentType: 'video/mp4;codecs="avc1.42E01E"'
+          }
+        ]
+      }
+    ];
+
+    const keySystemsMap = new Map([
+      ['widevine', 'com.widevine.alpha'],
```

Comment:
the list should be in consts.
and in a separate file, i think.
we shouldn't need to change kaltura-player.js every time we add a new type, only when we change the logic of the function.